lockfile_generator/src/pnpm.rs

```diff
@@ -25,3 +43,18 @@ impl Generator for Pnpm {
         "pnpm"
     }
 }
+
+/// Find PNPM workspace root.
+fn find_workspace_root(mut path: &Path) -> Option<PathBuf> {
+    loop {
+        let dir = fs::read_dir(path).ok()?;
+
+        for dir_entry in dir.into_iter().flatten().map(|entry| entry.path()) {
+            if dir_entry.file_name().map_or(false, |name| name == WORKSPACE_MANIFEST_FILENAME) {
+                return Some(path.into());
+            }
+        }
+
+        path = path.parent()?;
+    }
```

Comment:
You could consider using `path.ancestors()` here instead.

```rust
for path in path.ancestors() {
   // ...
}
None
```

It avoids `mut` and is perhaps slightly more descriptive.

I guess this entire function could even be `path.ancestors().find(...).map(PathBuf::from)`